content/en-US/api/CanvasRenderingContext2D.reset.md

```diff
@@ -0,0 +1,111 @@
+---
+recipe: api-interface
+title: 'CanvasRenderingContext2D.reset'
+mdn_url: /en-US/docs/Web/API/CanvasRenderingContext2D/reset
+specifications: [[https://html.spec.whatwg.org/multipage/canvas.html]]#[[dom-context-2d-reset]]
+browser_compatibility: api.CanvasRenderingContext2D.reset
+---
+
+
+**When this feature ships, the content below will live on MDN under
+[developer.mozilla.org/en-US/docs/Web/CanvasRenderingContext2D/reset](https://developer.mozilla.org/en-US/docs/Web/CanvasRenderingContext2D/reset).**
+
+## Description
+
+The `reset()` method of the `CanvasRenderingContext2D` interface
+resets the rendering context to its default state. This includes setting the backing buffer to transparent black, clearing the saved state stack, clearing the default path and resetting the drawing state to its initial values.
```

Comment:
This is well written. Terms like 'backing buffer' and 'saved state stack' concern me. Is this a description of what Chrome is doing under the hood? (This API is new to me.) MDN pages need to be written in terms of what web developers will observe when using the API.

Reply:
Would it be useful to define these terms? Backing buffer/back buffer/backing store is used here, alone with state stack:
https://html.spec.whatwg.org/multipage/canvas.html

The first one is problematic, I agree, as obviously no-one has decided what the best term is for "piece of computer memory that a canvas is drawing too that is usually pixels but not always"

State stack is more straighforward, as it is a "stack" (can we expect readers to know this term?) of "canvas drawing states" which are well defined:

https://html.spec.whatwg.org/multipage/canvas.html#drawing-state

Reply:
My main concern was keeping under the hood stuff out of developer reference. Sometimes these things turn out to be rabbit holes that nobody has time to get into. This looks like one of those times. Per MDN practice, the proper place for explanations full explanations of these is in the [guides and tutorials for the Canvas API](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API#guides_and_tutorials). There's a lot of material there and slapping something in if there's no natural place for it doesn't do much to help developer learn the material. 

Recommendation: link to materials that define these terms and call it good.

Reply:
Anyway, I think I get the gist of what it's saying, even without being familiar with those terms.

Reply:
Done! I went with "output bitmap" in place of "backing buffer" because that seems to be what's most commonly used in other documentation. Terms like "backing store", "buffer" and "stored bitmap are also used. This is all perhaps due for a refactor.